vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -29,17 +54,86 @@ public void setExhibition(JoinPoint jp) {
         Class<?> returnType = signature.getReturnType();
         // only if there is a model object injected and if we're returning a string
         // (assuming that returning a string implies rendering of a view afterwards
-        if (args != null && returnType == String.class) {
-            for (Object obj : args) {
-                if (obj instanceof Model && !(obj instanceof RedirectAttributes)) {
-                    if (!((Model) obj).containsAttribute("exhibition")) {
+        if(args!=null && returnType==String.class) {
+            for(Object obj : args) {
+                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if(!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
-                    if (!((Model) obj).containsAttribute("allSpaces")) {
+                    if(!((Model) obj).containsAttribute("allSpaces")) {
                         ((Model) obj).addAttribute("allSpaces", spaceManager.getAllSpaces());
                     }
                 }
+            }  
+        }
+    }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexofModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        if(exhibition==null) {
+            return "home";
+        }
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<String, String> ids = getIds(args, signature);
+        String spaceId = ids.getOrDefault("spaceId", "");
+        String moduleId = ids.getOrDefault("moduleId", "");
+        if(spaceId.equals("") && moduleId.equals("") && 
+            (exhibition.getMode().equals(ExhibitionModes.ACTIVE) || authFacade.getAuthenticatedUser()!=null)) {
+            return "notFound";
+        }
+        return redirectRequest(jp, spaceId, moduleId, indexofModel, exhibition);
+    }
+    
+    public Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String moduleId, int modelIndex, Exhibition exhibition) throws Throwable{
+        ISpace space = spaceManager.getSpace(spaceId);
+        IModule module = moduleManager.getModule(moduleId);
+        Object[] args = jp.getArgs();
+        if(exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            else {
+                return jp.proceed();
+            }
+        }
+        if(authFacade.getAuthenticatedUser()!=null && modelIndex>-1) {
+            if(space==null && module==null) {
```

Comment:
why do you check the modelIndex variable?

Reply:
also, how is that different from the previous times not found was returned?

Reply:
This checks if the user is not logged in and if logged in and incorrect space/module Ids, then we show notFound page. If Id is correct we show the exhibition but with a flag to show the pop-up message. Since Active is handled before this if-else, only offline or maintenance modes will apply here.

Reply:
since it doesn't matter if the user is logged in or not for the case that neither the space or the module exists, you should just have one check and return null.

Reply:
I don't think there is a reason to return 'notFound' more than once. You should be able to write the code so that it's very clear under which conditions it return not found, when it simply proceeds, and when it shows an alert.

Reply:
For Offline or Maintenance, it matters if the user is logged in or not, If they are and the module/space does not exist, we show does not exist page, if they are not logged in and space/module does not exist we still show offline / maintenance page.

Reply:
but if the exhibition is active, it doesn't matter.

Reply:
now I see, I think you just need to clean up your code and add some comments. One if statement that returns notfound (e.g. is active or is logged in and space == null or module == nul)